expo/expo
# [ios][android][background-task] add new package: expo-background-task

# Why

`expo-background-fetch` uses deprecated APIs and we'd like to replace these on both iOS and Android using the latest and greatest APIs. With this new package we can deprecate the `expo-background-fetch` package and move on.

The new APIs are [`BGTaskScheduler`](https://devhttps://developer.apple.com/documentation/backgroundtasks/bgtaskscheduler) API on iOS and the [`WorkManager`](https://developer.android.com/topic/libraries/architecture/workmanager) API on Android.

These are the recommended APIs for background tasks.

# How

- Added expo-background-task package
- Added documentation for the background-task package
- Added test screen in BareExpo
- Added necessary UIBackgroundModes in Expo-go and Bare-Expo

# Test Plan

- Run in BareExpo on device (iOS) or simulator (android) and open the BackgroundTask screen
- Test registration/unregistration
- Test running tasks
  - Test running tasks only runs the `background-task` (see log output)
  - Ensure that tasks are periodically repeated

# Checklist
